sim/dex-data.ts

```diff
@@ -341,54 +302,42 @@ export class Format extends BasicEffect {
 	maxForcedLevel?: number;
 	noLog: boolean;
 
-	constructor(data: AnyObject, moreData: AnyObject | null = null, moreData2: AnyObject | null = null) {
-		super(data, moreData, moreData2);
-		// @ts-ignore
-		this.mod = Tools.getString(this.mod) || 'gen7';
-		// @ts-ignore
+	constructor(data: AnyObject, ...moreData: (AnyObject | null)[]) {
+		data = combine(data, ...moreData);
+		super(data);
+		Object.assign(this, data); // TODO
+
+		this.mod = Tools.getString(data.mod) || 'gen7';
+		// @ts-ignore: this.effectType is assigned in super
```

Comment:
Typescript complains about `this.name` :(. It also complains the `Tool.getString(this.effectType)` could be anything, which is true, but `this.name` definitely seems like it would be assigned to me.

Reply:
Where does it complain about `this.name`? I don't see it being read anywhere nearby.

Reply:
Oops, the `this.name`comment was meant to be below, line 617.

Reply:
Where is line 617 on the latest change? I still can't find it.

Reply:
Oh, I fixed it with my `data = this` trick instead.